File: Regression/Section 6 - Polynomial Regression/Polynomial_Regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri Feb  7 18:06:01 2020

@author: kumet
"""


#importing libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd 

#importing the dataset
dataset = pd.read_csv("Position_Salaries.csv")
X = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values
"""
#Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size = 0.3,random_state = 0)

#Feature Scaling is done by the library
"""

#Fitting Linear Regression to the dataset
from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X,y)

#Fitting Polynomial Regression to the dataset
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly = poly_reg.fit_transform(X)
lin_reg_2 = LinearRegression()
lin_reg_2.fit(X_poly, y)


#Visualising the Polynomial Regression results
X_grid = np.arange(min(X),max(X),0.1)
X_grid = X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color="red")
plt.plot(X_grid,lin_reg_2.predict(poly_reg.fit_transform(X_grid)),color="blue")
plt.title("truth or bluff (Polynomial regression)")
plt.xlabel("Position Level")
plt.ylabel("Salary")
plt.show


#Visualising the Linear Regression results
plt.scatter(X,y,color="red")
plt.plot(X,lin_reg.predict(X),color="blue")
plt.title("truth or bluff (Linear regression)")
plt.xlabel("Position Level")
plt.ylabel("Salary")
plt.show


#Predicting a new result with Linear Regression
lin_reg.predict([[6.5]])
#Prediciting a new result with Polynomial Regression
lin_reg_2.predict(poly_reg.fit_transform([[6.5]]))

File: Regression/Section 6 - Polynomial Regression/Polynomial_Regression.R
dataset = read.csv("Position_Salaries.csv")
dataset = dataset[,2:3]

lin_reg = lm(formula = Salary ~., data=dataset)

dataset$Level2 = dataset$Level ** 2;
dataset$Level3 = dataset$Level ** 3;
dataset$Level4 = dataset$Level ** 4;
dataset$Level5 = dataset$Level ** 5;
dataset$Level6 = dataset$Level ** 6;
dataset$Level7 = dataset$Level ** 7;
dataset$Level8 = dataset$Level ** 8;
dataset$Level9 = dataset$Level ** 9;


poly_reg = lm(formula = Salary ~., data=dataset)

library(ggplot2)
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color ="red")+
  geom_line(aes(x=dataset$Level, y = predict(lin_reg, newdata= dataset)),color = "blue")+
  geom_line(aes(x=dataset$Level, y = predict(poly_reg,newdata=dataset)),color="green")+
  ggtitle("truth or bluff")+xlab("Level")+ylab("Salary")

#predicting a new result with Linear Regression
y_pred = predict(lin_reg, data.frame(Level=6.5))
y_pred2 = predict(poly_reg,data.frame(Level=6.5,
                                      Level2 = 6.5**2,
                                      Level3 = 6.5**3,
                                      Level4 = 6.5**4,
                                      Level5 = 6.5**5,
                                      Level6 = 6.5**6,
                                      Level7 = 6.5**7,
                                      Level8 = 6.5**8,
                                      Level9 = 6.5**9))
